types/supercluster/supercluster-tests.ts

```diff
@@ -1,23 +1,66 @@
 import supercluster, { Point } from 'supercluster';
 
+interface TestMapProperties {
+    firstLetterOfName: string;
+    lengthOfName: number;
+}
+
+interface TestReduceProperties {
+    aggregateFirstLetters: string;
+    aggregateLengthOfNames: number;
+}
+
+interface GeoJsonProperties {
+    name: string;
+}
+
 const point1: Point = {
     type: 'Feature',
-    properties: {},
+    properties: {name: "Bob"},
     geometry: { type: 'Point', coordinates: [10, 20] }
 };
 const point2: Point = {
     type: 'Feature',
-    properties: {},
+    properties: {name: "Ashley"},
     geometry: { type: 'Point', coordinates: [20, 30] }
 };
 const points = [point1, point2];
 
+const initialFunction: () => TestReduceProperties = () => {
+    return {
+        aggregateFirstLetters: "",
+        aggregateLengthOfNames: 0,
+    };
+};
+
+const mapFunction: (props: GeoJsonProperties) => TestMapProperties = (props) => {
+    if (!props.name || (props.name && props.name.length === 0)) {
+        return {
+            firstLetterOfName: "",
+            lengthOfName: 0
+        };
+    }
+    return {
+        firstLetterOfName: props.name.charAt(0),
+        lengthOfName: props.name.length,
+    };
+};
+
+const reduceFunction: (acc: TestReduceProperties, value: TestMapProperties) => void = (acc, value) => {
```

Comment:
I'd simplify these test examples since you are now using `any` and I'd try to avoid overly complicated Typescript test cases.


```ts
const initialFunction = () => {
  return 0
}
```


```ts
const mapFunction = (props) => {
  return props
}
```

```ts
const reduceFunction = (accumulated, props) => {
    if (props.name === 'Ashley') accumulated++; 
    return accumulated;
};
```

Reply:
sure I'll just do the sum accumulated